src/software/pipeline/main_imageMatching.cpp

```diff
@@ -459,135 +544,135 @@ int main(int argc, char** argv)
       ALICEVISION_LOG_INFO(ss.str());
     }
 
-    // create the database
-    ALICEVISION_LOG_INFO("Creating the database...");
+    // create the databases
+    ALICEVISION_LOG_INFO("Creating the databases...");
 
     // add each object (document) to the database
-    aliceVision::voctree::Database db(tree.words());
+    aliceVision::voctree::Database dbA(tree.words());
+    aliceVision::voctree::Database dbB;
+    aliceVision::voctree::Database dbAB;
```

Comment:
Having 3 databases make things complicated.

You only need a 2nd database in *the* particular case A_A_AND_A_B. In all other cases, you just need to fill the DB according to the needs, and all the other code is the same.

Reply:
I personally prefer the 3 databases way. Found it more generic and comprehensible.
But I did it in a next commit.